ReactiveObjCBridge/ObjectiveCBridging.swift

```diff
@@ -101,119 +112,200 @@ extension QueueScheduler {
-	return _bridgedSignalProducer(from: signal, file: file, line: line)
+	/// Create a `RACScheduler` that wraps the given scheduler.
+	///
+	/// - parameters:
+	///   - scheduler: The `DateScheduler` to wrap.
+	///
+	/// - returns: A `RACScheduler` that schedules blocks to `scheduler`.
+	public convenience init(_ scheduler: DateScheduler) {
+		self.init(racSwiftScheduler: RACSwiftScheduler(wrapping: scheduler))
+	}
 }
 
-private func _bridgedSignalProducer<Value>(from signal: RACSignal<Value>, file: String = #file, line: Int = #line) -> SignalProducer<Value?, AnyError> {
-	return SignalProducer<Value?, AnyError> { observer, disposable in
-		let next: (_ value: Value?) -> Void = { obj in
-			observer.send(value: obj)
+private final class RACSwiftScheduler: RACScheduler {
+	enum Backing {
+		case scheduler(Scheduler)
+		case dateScheduler(DateScheduler)
+	}
+
+	private let base: Backing
+
+	init(wrapping base: Scheduler) {
+		self.base = .scheduler(base)
+	}
+
+	init(wrapping base: DateScheduler) {
+		self.base = .dateScheduler(base)
+	}
+
+	private func wrap(_ block: @escaping () -> Void) -> () -> Void {
+		return {
+			Thread.current.threadDictionary["RACSchedulerCurrentSchedulerKey"] = self
+			block()
+			Thread.current.threadDictionary["RACSchedulerCurrentSchedulerKey"] = nil
 		}
+	}
+
+	open override func schedule(_ block: @escaping () -> Void) -> RACDisposable? {
+		switch base {
+		case let .scheduler(scheduler):
+			return scheduler.schedule(wrap(block)).map(RACDisposable.init)
 
-		let failed: (_ error: Swift.Error?) -> Void = { error in
-			observer.send(error: AnyError(error ?? defaultNSError("Nil RACSignal error", file: file, line: line)))
+		case let .dateScheduler(scheduler):
+			return scheduler.schedule(wrap(block)).map(RACDisposable.init)
 		}
+	}
+
+	open override func after(_ date: Date, schedule block: @escaping () -> Swift.Void) -> RACDisposable? {
+		switch base {
+		case let .scheduler(scheduler):
+			Thread.sleep(until: date)
+			return scheduler.schedule(wrap(block)).map(RACDisposable.init)
 
-		let completed = {
-			observer.sendCompleted()
+		case let .dateScheduler(scheduler):
+			return scheduler.schedule(after: date,
+			                          action: wrap(block)).map(RACDisposable.init)
 		}
+	}
 
-		disposable += signal.subscribeNext(next, error: failed, completed: completed)
+	open override func after(_ date: Date, repeatingEvery interval: TimeInterval, withLeeway leeway: TimeInterval, schedule block: @escaping () -> Void) -> RACDisposable? {
+		switch base {
+		case let .scheduler(scheduler):
+			assertionFailure("Undefined behavior.")
+			return scheduler.schedule(wrap(block)).map(RACDisposable.init)
+
+		case let .dateScheduler(scheduler):
+			return scheduler.schedule(after: date,
+			                          interval: .milliseconds(Int(interval * 1000)),
+			                          leeway: .milliseconds(Int(leeway * 1000)),
```

Comment:
You should be able to do `.seconds(Int(interval))` instead of multiplying into milliseconds?